src/promptflow/promptflow/_core/tracer.py

```diff
@@ -358,10 +373,12 @@ async def wrapped(*args, **kwargs):
                 Tracer.push(trace)
                 enrich_span_with_input(span, trace.inputs)
                 output = await func(*args, **kwargs)
-                if trace_type == TraceType.LLM:
+                if trace_type in [TraceType.LLM, TraceType.EMBEDDING]:
                     token_collector.collect_openai_tokens(span, output)
                 enrich_span_with_output(span, output)
                 enrich_span_with_openai_tokens(span, trace_type)
+                if trace_type == TraceType.EMBEDDING:
+                    enrich_span_with_embedding(span, trace.inputs, output)
```

Comment:
We may have something like `enrich_span_by_type(span, inputs, output, trace_type)` to avoid if else here?

Reply:
Refined.